sim-cli/src/main.rs

```diff
@@ -199,3 +217,47 @@ async fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+async fn read_sim_path(data_dir: PathBuf, sim_file: String) -> anyhow::Result<PathBuf> {
+    let sim_path = data_dir.join(sim_file);
+
+    let sim_path = if sim_path.extension().is_none() {
+        sim_path.with_extension("json")
```

Comment:
Kind of like the same as with the file picker. Do we want to assume the user won't be able to specify the file extension?

IMO this whole function shouldn't be needed

Reply:
This ergonomic allows user to specify file name without extension.
The rest of the function just builds a 'nice UX' around the `data_dir` + `sim_file` pairing, so teh question is whether users need that UX or not

Reply:
Sure, I just find whoever is using sim-ln should be able to properly define a path and a file name with an extension. It is nice to have UX features that make it easy for the user to run the software, but I find this is way too much. We're expecting the user to be able to define the sim file which already contains paths, extensions and so on

Reply:
💯 now I see how it's over-polished